frontend/spec/controllers/spree/checkout_controller_spec.rb

```diff
@@ -170,10 +170,59 @@ def spree_post_address
         end
       end
 
+      # This is the only time that we need the 'set_payment_parameters_amount'
+      # controller code, because otherwise the transition to 'confirm' will
+      # trigger the 'add_store_credit_payments' transition code which will do
+      # the same thing here.
+      # Perhaps we can just remove 'set_payment_parameters_amount' entirely at
+      # some point?
+      context "when there is a checkout step between payment and confirm" do
+        before do
+          @old_checkout_flow = Spree::Order.checkout_flow
+          Spree::Order.class_eval do
+            insert_checkout_step :new_step, after: :payment
+          end
+        end
+
+        after do
+          Spree::Order.checkout_flow(&@old_checkout_flow)
+        end
+
+        let(:order) { create(:order_with_line_items) }
```

Comment:
This regression test was added before `OrderUpdateAttributes` was created. It provides us value in testing a corner case that the original code accounted for. For this reason it provides value.